pkg/mtu/detect_linux.go

```diff
@@ -75,3 +75,33 @@ func autoDetect() (int, error) {
 
 	return EthernetMTU, nil
 }
+
+// getMTUFromIf finds the interface that holds the ip and returns its mtu
+func getMTUFromIf(ip net.IP) (int, error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return 0, fmt.Errorf("Unable to list interfaces: %s", err)
+	}
+
+	for _, iface := range ifaces {
+		addrs, err := iface.Addrs()
+		if err != nil {
+			return 0, fmt.Errorf("Unable to list the addresses: %s", err)
+		}
+
+		for _, addr := range addrs {
+			myIP, _, err := net.ParseCIDR(addr.String())
+
+			if err != nil {
+				return 0, fmt.Errorf("Unable to parse the CIDR: %s", err)
```

Comment:
Same comment as above, presumably it's better to skip cases we can't handle than to error out on the MTU detection entirely?

Reply:
Ok, let me redo this and in case of error move to what we had before this patch